compendium/ServiceComponent/include/cppmicroservices/servicecomponent/usrDefinedMethodAssertion.hpp

```diff
@@ -0,0 +1,50 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  =============================================================================*/
+
+#ifndef usrDefinedMethodAssertion_hpp
+#define usrDefinedMethodAssertion_hpp
+
+#include "cppmicroservices/servicecomponent/ComponentContext.hpp"
+
+namespace cppmicroservices::service::component
+{
+    template <typename T>
+    struct usrDefinedMethodAssertion
```

Comment:
I hate to be pedantic, but I believe that there's a "coding standard" at MathWorks that says that class names should begin with an upper case letter. Also, the "abbreviation" of 1 letter, usr instead of User... is it worth it to be possibly less clear to save one character?